_posts/2023/2023-06-06-getting-started-with-vector-dbs-in-python.md

```diff
@@ -376,6 +376,151 @@ ENDPOINT=http://localhost:8080 CONFIG_ENDPOINT=http://localhost:19071 poetry run
 {% endhighlight %}
 
+### Weaviate
+Weaviate is a vector search engine specifically designed for natural language and numerical data. It uses contextualized embeddings in data objects to understand semantic similarity. Currently, it supports only Hierarchical Navigable Small World(HNSW) indexing and is more costly on building data to indexes however has a fast query time and high scalability. Weaviate is open-source, easy to use, flexible, extensible and has a Contributor License Agreement.
+
+After you sign up at Weaviate Cloud Services WCS, and create a new free tier Weaviate Cluster with authentication. Note your cluster URL and API key (optional). The endpoint will have the following format https://myindex.weaviate.network.
+
+It is easy to create some objects with vectors.
+
+{% highlight python %}
+
+with Client(
+    event_hooks={
+        "request": [log_request],
+        "response": [log_response, Response.raise_for_status],
+    }
+) as client:
+    # index data
+    vectors = [
+        {
+            "id": "vec1",
+            "values": [0.1, 0.2, 0.3],
+            "properties": {
+                "genre": "drama"
+            }
+        },
+        {
+            "id": "vec2",
+            "values": [0.2, 0.3, 0.4],
+            "properties": {
+                "genre": "action"
+            }
+        }
+    ]
+
+    objects = []
+    for vector in vectors:
+        obj = {
+            "class": "Vectors",
+            "properties": {
+                "vector": vector["values"]
+            }
+        }
+        objects.append(obj)
+
+    client.post(urljoin(endpoint, "/v1/batch/objects"), json={"objects": objects}, headers=headers)
+{% endhighlight %}
+
+The search is pretty straight forward. Weaviate also has a GraphQl interface online.
+
+{% highlight python %}
+    # search for data
+    query = {
+        "fields": "vector",
+        "nearVector": {
+            "vector": [0.1],
+            "certainty": 0.9
+        }
+    }
+
+    response = client.get(urljoin(endpoint, "/v1/objects"), params=query, headers=headers).json()
+    for obj in response["objects"]:
+        print(obj)
+{% endhighlight %}
+
+Deleting objects of the same class is simple.
+
+{% highlight python %}
+    client.delete(urljoin(endpoint, f"/v1/schema/Vectors"), headers=headers)
+{% endhighlight %}
+
+You can see and run a [working sample from here](https://github.com/dblock/vectordb-hello-world/blob/main/src/weaviate/hello.py).
+
+
+### Qdrant
+Qdrant is similarity vector search engine designed for a wide range of applications, including recommendation systems, image search, and natural language processing. It is open-source, scalable and allows dynamic updates to the index. It is particularly suitable for scenarios where the vector data is constantly evolving and vectors may be modified without interrupting the search functionality. Quadrant is licensed under Apache 2.0.
```

Comment:
Add a line, link Qdrant.

You say open-source and licensed under APLv2, it's the same thing. Would just leave one.